File: src/js/katas/7kyu/credit_card_mask.js
/* Kata url: https://www.codewars.com/kata/5412509bd436bd33920011bc. */

function maskify(cc) {
  let len = cc.length;

  if (len <= 4) {
    return cc;
  }

  return "#".repeat(len - 4) + cc.substring(len - 4);
}

module.exports = maskify;

File: src/python/katas/py7kyu/credit_card_mask.py
"""Kata url: https://www.codewars.com/kata/5412509bd436bd33920011bc."""


def maskify(cc: str) -> str:
    return cc[-4:].rjust(len(cc), "#")


def test_maskify():
    assert maskify("") == ""
    assert maskify("1") == "1"
    assert maskify("64607935616") == "#######5616"
    assert maskify("4556364607935616") == "############5616"
    assert maskify("SF$SDfgsd2eA") == "########d2eA"
